pylib/Tools/Build/Shell.py

```diff
@@ -362,6 +369,7 @@ def execute(self, log, keyvals, testDef):
                 log['status'] = results['status']
```

Comment:
line 333 add
log['stderr'] = 'Failed to Allocate Resources'
log['status'] = 1
log['result'] = testDef.MTT_TEST_FAILED
os.chdir(cwd)

line 358 ??? do we really want to exit here? Maybe just print a verbose message saying that this happened.  The resource manager should be able to handle this by timeout.  If we really do want to return then the following needs to be added.
log['status'] = 1
log['result'] = MTT_RESULT_line 333 add
log['stderr'] = 'Failed to Allocate Resources'
log['status'] = 1
log['result'] = testDef.MTT_TEST_FAILED
os.chdir(cwd)

On line 360 it would be helpful to have a comment describing what this if/else is for. Looks like it is checking if the test failed and exits if it did. Lines 370 and 371 need to be indented. and line 372 needs to be changed to failed
log['result'] = testDef.MTT_TEST_FAILED

Add log['result'] = testDef.MTT_TEST_PASSED to line 380

Reply:
I have no idea - I didn't write it, or at least I sure don't recall doing so. I'd rather leave the logic as-is and add only what is minimally required.

Lines 370,371 do not need to be indented - they are correct as-is. The logic is obtuse, but the general idea appears to be that the test either reported success when it should have failed (line 366), or it failed but returned the incorrect status.